### ISSUE TITLE
Could not load type Mediatr.IRequest

### ISSUE THREAD

**Issue by bend** [jbogard/MediatR#692]
Hi, 
Since update to Mediatr 10, I have the following error at runtime: 
` Could not load type 'MediatR.IRequest 1' from assembly 'MediatR, Version=10.0.0.0, Culture=neutral, PublicKeyToken=bb9a41a5e8aaa7e2' `

Any idea what's going on ?

**Comment by vRITHNER** [comment]
Same issue for me with an AspNet.Core 6.0 app

**Comment by jbogard** [comment]
I've got it running on .NET 6, is there a repro? Are you using the DI package?

**Comment by jbogard** [comment]
And the exception message is correct - those types are in a separate MediatR.Contracts assembly as mentioned in the release notes. But user code shouldn't care about that in scenarios I looked at.

**Comment by vRITHNER** [comment]
txs for coming back. Sadly, I can't send you the project cause it's a customer code.
I'll try to reproduce it in a total separate poc project and I'll come back to you..
Txs

**Comment by jbogard** [comment]
If you check your stack trace, do you have code looking for the `IRequest` type from the MediatR assembly?

**Comment by bend** [comment]
@jbogard  I don't see that in the stacktrade. However, I have the issue in multiple projects.
  [CS7069] Reference to type 'INotification' claims it is defined in 'MediatR', but it could not be found

I tried to force the update by install Mediatr10 in the project itself but I have the same issue.
Thanks

**Comment by jbogard** [comment]
I don't know how to fix this without a minimal repro. I can only share that those request types are in a separate package/assembly - `MediatR.Contracts`, which is referenced by the `MediatR` package starting with version 10.

**Comment by killergege** [comment]
Note that I have the same issue, but it happens when calling `MediatR.Extensions.FluentValidation.AspNetCore.AddFluentValidation()`.
Which is an external lib. 
The update probably shouldn't break that lib though...

Nobody mentions here what triggers the exception, so maybe we all have the same issue ?

In my case, the exception happen on the second line.
```
        services.AddMediatR(typeof(MyRequestHandler).GetTypeInfo().Assembly);
        services.AddFluentValidation(new[] { typeof(MyRequestHandlerValidator).GetTypeInfo().Assembly });
```
Removing it fixes the issue.

I have a very stripped down version of my project with the issue, but if the issue is in another lib, not sure this is interesting.

**Comment by DMPatod** [comment]
I'm using dotnet 6 a had to drop to MediatR 8.1 and MediatR.Extensions.Microsoft.DependencyInjection 8.1 to solve this.
Neither 10.0.0 and 9.0.0 worked for me.

**Comment by dhilgarth** [comment]
@jbogard Also coming across this. My scenario:
a Nuget package, on .NET 5 and MediatR 9.0.0 is referenced by a class library on .NET 6 and MediatR 10.0.1.
The Nuget package contains a command that implements `IRequest`. Trying to `mediator.Send(new CommandFromPackage())` will result in this error on the Send method.

**Comment by aboudoux** [comment]
Hi,
Same problem here. 
everything was fine until I include Blazor-State in my project.

Downgrading the solution to MediatR 9 resolve the issue for me.

**Comment by jbogard** [comment]
The assemblies were split out and `IRequest` is in a separate assembly. If things are referencing both older and newer versions, things are going to break. Anything referencing MediatR will need to update major versions.

This is pretty normal for major versions, you can't mix major versions of assemblies without expecting some failures.

**Comment by ydie22** [comment]
Hi,

I just opened a pull request that should solve the issue described here. See https://github.com/jbogard/MediatR/pull/800.

I understand that major upgrades contain breaking changes, however this one is particularly hard to solve for us. We have tens of contracts packages that are currently referring to MediatR 8.1 (where the interfaces for request types were living until then), and those are consumed by many services. We cannot afford upgrading everything at once. I think the proposed change will allow upgrading a service to the latest MediatR version while still referencing contract packages that consume MediatR 8.1 (provided of course there are no other breaking changes than this one that affect the code base).

Thx

**Comment by jbogard** [comment]
Fixed by #800

**Comment by chrisira** [comment]
I solved the problem by downgrading from mediatr 11.10 t0 11.0
